tools/lsp/preview/ui/property_view.rs

```diff
@@ -178,19 +180,166 @@ fn convert_number_literal(
+        }
+        expression_tree::Expression::StoreLocalVariable { value, .. } => {
+            println!("Stripping StoreLocalVariable");
+            return extract_color_from_evaluated_expression_impl(value, palette_name, struct_level);
+        }
+        _ => {
+            eprintln!("NOT HANDLED: {evaluated_expression:?}");
+        }
+    };
+    ExtractedColor::default()
+}
+
+fn extract_color_from_evaluated_expression(
+    evaluated_expression: &expression_tree::Expression,
+) -> ExtractedColor {
+    extract_color_from_evaluated_expression_impl(evaluated_expression, Vec::new(), 0)
+}
+
 fn extract_color(
-    expression: &syntax_nodes::Expression,
+    expression: &Option<syntax_nodes::Expression>,
+    evaluated_expression: &Option<expression_tree::Expression>,
     kind: ui::PropertyValueKind,
     value: &mut ui::PropertyValue,
 ) -> bool {
-    if let Some(text) = expression.child_text(SyntaxKind::ColorLiteral) {
-        if let Some(color) = ui::string_to_color(&text) {
-            value.display_string = text.as_str().into();
-            value.kind = kind;
-            value.value_brush = slint::Brush::SolidColor(color);
-            value.gradient_stops =
-                Rc::new(VecModel::from(vec![ui::GradientStop { color, position: 0.5 }])).into();
-            return true;
+    if let Some(expression) = expression {
+        if expression.children_with_tokens().any(|child| {
+            eprintln!("CODE TEST: Child kind: {:?}", child.kind());
+            ![SyntaxKind::QualifiedName, SyntaxKind::ColorLiteral].contains(&child.kind())
+        }) {
+            eprintln!("CODE IS TOO COMPLEX TO BE CONSIDERED");
+            return false;
+        }
+    }
+    if let Some(ev) = evaluated_expression {
+        let color = extract_color_from_evaluated_expression(ev);
+        match color {
+            ExtractedColor { value: None, name: None } => {
+                // leave this as code...
+            }
+            ExtractedColor { value: color, name } => {
+                let value_string = name
+                    .clone()
+                    .unwrap_or_else(|| {
+                        let expression_string =
+                            expression.as_ref().map(|e| e.text().to_string()).unwrap_or_default();
+                        let expression_string = expression_string.trim();
+
+                        let color_name =
+                            expression_string.strip_prefix("Colors.").unwrap_or(expression_string);
+                        match i_slint_compiler::lookup::named_colors().get(color_name) {
+                            Some(c) if *c == color.unwrap_or_default().as_argb_encoded() => {
+                                format!("Colors.{color_name}")
+                            }
+                            _ => String::new(),
+                        }
+                    })
+                    .into();
+                value.display_string = name
+                    .clone()
```

Comment:
maybe something like
```suggestion
                    .as_ref()
```

to avoid double allocation  (cloning just to construct a SharedString out of it)

Reply:
Nice! Will be in the next push.